### PR TITLE
Use ScalaTest 3.1 for derpication warnings

### PR DESCRIPTION
**Pull request by schoeberl**
not yet to merge, needs to be tested

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -70,7 +70,7 @@ scalacOptions ++= Seq(
 libraryDependencies ++= Seq(
   "edu.berkeley.cs" %% "chisel3"       % defaultVersions("chisel3"),
   "edu.berkeley.cs" %% "treadle"       % defaultVersions("treadle"),
-  "org.scalatest"   %% "scalatest"     % "3.2.9",
+  "org.scalatest"   %% "scalatest"     % "3.1.2",
```
Comment (ekiwi) [inline build.sbt:73]: `3.1.4` seems to be the latest `3.1.x` release. Could you try changing it to that?

**Review by ekiwi — CHANGES_REQUESTED** [review]
Thanks for paying attention and looking into this @schoeberl !
I agree that making users switch from 3.0 to 3.2 directly without the deprecation warnings would have been a bad idea.

File: build.sbt (hunk 1 above)
Comment (schoeberl) [reply build.sbt]: 3.1.4 works as well, did do the change

**Review by ekiwi — APPROVED** [review]
Thanks! :ship:

**Comment by sequencer** [comment]
Basically, It is troublesome to force ChiselTest depending on ScalaTest, this will force user codes depends on the same version as ChiselTest does, user cannot change or bump their ScalaTest version.
I think in the future, better way might be spliting ScalaTest and UTest dependency out from ChiselTest codebase to decouple this dependency, Currently ScalaTest API only creates test folder and wrapper the exception. maybe it might be better to split them out from ChiselTest codebase.

**Comment by ekiwi** [comment]
I agree with splitting out utest. However, for non-expert users, I think it is better to provide a default solution that works with a wide range of IDEs + sbt.

However, one thing we could consider is to actually inherit a specific spec.
`scalatest` actually recommends that you do not extend `AnyFlatSpec` for every test you write. Instead you are supposed to make one project specific spec and then use that everywhere. If we had done that, we wouldn't have to change things in quite so many places. Intel did the right thing for VTA: https://github.com/apache/tvm-vta/blob/main/hardware/chisel/src/test/scala/unittest/Generic.scala#L30

So one way to help newcomers while sheltering them from `scalatest` changes would be to define our own `ChiselSpec` that directly inherits from `AnyFlatSpec` and then change the docs to tell people to always use the `ChiselSpec`.

**Pull request merged**